src/ape/api/transactions.py

```diff
@@ -200,6 +203,56 @@ def decode_logs(self, abi: Union[EventABI, "ContractEvent"]) -> Iterator[Contrac
 
         yield from self.provider.network.ecosystem.decode_logs(abi, self.logs)
 
+    @property
+    def events(self):
+        contracts = {log["address"] for log in self.logs}
+        contract_types = {addr: self.chain_manager.contracts.get(addr) for addr in contracts}
+        decoded = []
+        for log in self.logs:
+            note = self.decode_ds_note(contract_types[log["address"]], log)
```

Comment:
To get this working with decode_logs, we need to make it more generic, and pass it the full ContractType.